woocommerce/woocommerce
# Added counties/states for EU countries and GB.

This should fix the problematic behaviour of automated taxes when using WooCommerce Services, Jetpack and TaxJar + WooCommerce.

### All Submissions:




### TODO:
* [ ] check that matching of shipping zones still works correctly, I saw some strange effects where it didn't match any shipping zone, even the catch-all.



### Changes proposed in this Pull Request:



Closes #24354 .
Counties and subdivisions coming from CLDR release v36: http://unicode.org/Public/cldr/36/
Where available, I used the local state/county names. 

### How to test the changes in this Pull Request:

1. Install WooCommerce with WooCommerce Services, connect Jetpack and set automated tax calculations.
2. Try to check out a couple of times using different County/State, e.g. for Country GB, try empty, multiple word and one word options.
3. Check that tax is only applied if the County/State does not have a space in it.
4. Apply branch
5. Test checkouts again, it should work fine in all cases. A dropdown with County selection should appear instead of a text input field, but it should **not** be compulsory to fill in any of the options.
6. Verify that the automatically created tax rate rules are no longer duplicated.
7. Remove all the automatically created tax rates, add manual ones and verify the checkout calculates the tax correctly.

### Other information:




### Changelog entry

> Fix - Added counties/states for EU countries and GB to fix edge case with automated tax calculation.
